docs/map.md

```diff
@@ -0,0 +1,172 @@
+# map block
+
+The `map` block can be used to declare complex objects inside 
+[Globals](sharing-data.md) and it's useful when aggregating or merging lists
+of objects that potentially can have duplicated keys.
```

Comment:
```suggestion
The `map` block can be used to create complex maps/objects inside 
[Globals](sharing-data.md). It can be used to aggregate lists
of objects into maps that have duplicated keys and need a defined way of deep merging values of the same key.
```